WcaOnRails/app/webpacker/lib/show-schedule.js

```diff
@@ -10,6 +10,44 @@ window.wca.registerVenueData = (id, venueData) => {
   dataByVenueId[id] = venueData;
 };
 
+const HEX_BASE = 16;
+
+/**
+ * Convert a HEX color code to RGB values.
+ *
+ * @param {string} hexColor HEX color code to convert to RGB
+ *
+ * @returns {Array<number>} RBG values, defaults to `[0, 0, 0]` if `hexColor` cannot be parsed
+ */
+const hexToRgb = (hexColor) => {
+  if (/#[0-9A-Fa-f]{6}/.test(hexColor)) {
+    return [
+      parseInt(hexColor.slice(1, 3), HEX_BASE),
+      parseInt(hexColor.slice(3, 5), HEX_BASE),
+      parseInt(hexColor.slice(5, 7), HEX_BASE),
+    ];
+  }
+
+  return [0, 0, 0];
+};
+
+const WHITE = '#ffffff';
+const BLACK = '#000000';
+
+/**
+ * Compute appropriate text color (black or white) based on how "light" or "dark"
+ * the background color of a calendar item is.
+ *
+ * @param {string} backgroundColor Calendar item's background color (in HEX)
+ *
+ * @returns {string} white for "dark" backgrounds, black for "light" backgrounds
+ */
+const getTextColor = (backgroundColor) => {
+  const [red, green, blue] = hexToRgb(backgroundColor);
```

Comment:
can you add a toLowerCase() to backgroundColor because it is not clear for the user (if they want to use this function in any other code) if the Color is supposed to be using lower or uppercase letters (I just did it wrong testing it)

Reply:
Sure, I'll add `.toLowerCase()` to makes things clear, but I'm curious if you have an example of the function breaking?

The regex in `hexToRgb` handles lowercase characters and `parseInt` interprets uppercase and lowercase the same way:

```js
> const HEX_BASE = 16;
undefined
> const hexToRgb = (hexColor) => {
...   if (/#[0-9A-Fa-f]{6}/.test(hexColor)) {
.....     return [
.....       parseInt(hexColor.slice(1, 3), HEX_BASE),
.....       parseInt(hexColor.slice(3, 5), HEX_BASE),
.....       parseInt(hexColor.slice(5, 7), HEX_BASE),
.....     ];
.....   }
...
...   return [0, 0, 0];
... };
undefined
> hexToRgb("#Fa8bCd");
[ 250, 139, 205 ]
> hexToRgb("#FA8CCD");
[ 250, 140, 205 ]
> hexToRgb("#fa8ccd");
[ 250, 140, 205 ]
> parseInt("FC", HEX_BASE);
252
> parseInt("fc", HEX_BASE);
252
> parseInt("Fc", HEX_BASE);
252
>
```

Reply:
Ah sorry, I tested it like this in the console
```
const colors = ["#FFFFFF", "#AAAAAA", "#99AD32"]
for (color in colors){
console.log(getTextColor(color))
}
```
and made the error that color is not the value but the array index... 
So it's actually correct, sorry for that